### PR TITLE
List supported operators in hardware requirement docs

### PR DESCRIPTION
**Pull request by happz**
Regex operators were documented, others were mentioned in examples. Let's add a paragraph just for them.

### CODE REVIEW

File: spec/hardware/main.fmf [hunk 1]
```diff
@@ -19,6 +19,26 @@ description: |
     grouped under a common parent, for example ``cores`` or
     ``model`` under the :ref:`/spec/hardware/cpu` key.
 
+    Operators
+    ---------
+
+    The ``value`` part of the constraint follows the schema ``[operator] actual-value``.
```
Comment (thrix) [inline spec/hardware/main.fmf:25]: Could we mention it is advised to quote the constraint and the value? And is it?

File: spec/hardware/main.fmf [hunk 2]
```diff
@@ -19,6 +19,26 @@ description: |
     grouped under a common parent, for example ``cores`` or
     ``model`` under the :ref:`/spec/hardware/cpu` key.
 
+    Operators
```
Comment (thrix) [inline spec/hardware/main.fmf:22]: `Value comparison operators` maybe? Rather avoid the generic `Operators` naming?

**Review by thrix — CHANGES_REQUESTED** [review]

File: spec/hardware/main.fmf (hunk 2 above)
Comment (janhavlin) [reply spec/hardware/main.fmf:22]: `Operators` seem obvious enough to me...

**Review by janhavlin — APPROVED** [review]

File: spec/hardware/main.fmf (hunk 2 above)
Comment (happz) [reply spec/hardware/main.fmf:22]: Same here.

Comment (happz) [reply spec/hardware/main.fmf:22]: Let's add "Comparison" - we already have "Logic" operators.

File: spec/hardware/main.fmf (hunk 1 above)
Comment (happz) [reply spec/hardware/main.fmf:25]: I'm not sure we can enforce it, it's behind a YAML and if the user omits quotes, tmt would never know, but we can definitely recommend using them. Added in 69169daa.

File: spec/hardware/main.fmf (hunk 2 above)
Comment (happz) [reply spec/hardware/main.fmf]: Changed in 69169daa.

File: spec/hardware/main.fmf (hunk 1 above)
Comment (lukaszachy) [reply spec/hardware/main.fmf:25]: @thrix yq is strange. other yaml parsers (e.g. ruamel):

```
YAML(typ='safe').load("memory: != 8 GB")
ConstructorError: could not determine a constructor for the tag '!='
```

Comment (thrix) [reply spec/hardware/main.fmf:25]: indeed ....

**Review by thrix — APPROVED** [review]

**Pull request merged**